components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/util/I18nEmailUtil.java

```diff
@@ -232,4 +245,203 @@ public static String prependOperationScenarioToErrorCode(String exceptionErrorCo
+                        I18nMgtConstants.ErrorMessages.ERROR_CODE_INVALID_EMAIL_TEMPLATE.getMessage());
+            }
+        }
+    }
+
+    /**
+     * Validate the display name of the notification template.
+     *
+     * @param displayName Display name
+     * @throws NotificationTemplateManagerClientException Invalid notification template name
+     */
+    public static void validateDisplayNameOfTemplateType(String displayName)
+            throws NotificationTemplateManagerClientException {
+
+        if (StringUtils.isBlank(displayName)) {
+            String errorCode =
+                    I18nEmailUtil.prependOperationScenarioToErrorCode(
+                            I18nMgtConstants.ErrorMessages.ERROR_CODE_EMPTY_TEMPLATE_NAME.getCode(),
+                            I18nMgtConstants.ErrorScenarios.EMAIL_TEMPLATE_MANAGER);
+            throw new NotificationTemplateManagerClientException(errorCode,
+                    I18nMgtConstants.ErrorMessages.ERROR_CODE_EMPTY_TEMPLATE_NAME.getMessage());
+        }
+        /*Template name can contain only alphanumeric characters and spaces, it can't contain registry invalid
+        characters*/
+        String[] whiteListPatterns = {TEMPLATE_REGEX_KEY};
+        String[] blackListPatterns = {REGISTRY_INVALID_CHARS};
+        if (!IdentityValidationUtil.isValid(displayName, whiteListPatterns, blackListPatterns)) {
+            String errorCode =
+                    I18nEmailUtil.prependOperationScenarioToErrorCode(
+                            I18nMgtConstants.ErrorMessages.ERROR_CODE_INVALID_CHARACTERS_IN_TEMPLATE_NAME.getCode(),
+                            I18nMgtConstants.ErrorScenarios.EMAIL_TEMPLATE_MANAGER);
+            String message =
+                    String.format(
+                            I18nMgtConstants.ErrorMessages.ERROR_CODE_INVALID_CHARACTERS_IN_TEMPLATE_NAME.getMessage(),
+                            displayName);
+            throw new NotificationTemplateManagerClientException(errorCode, message);
+        }
+    }
+
+    /**
+     * Validates the locale code of a notification template.
+     *
+     * @param locale Locale code
+     * @throws NotificationTemplateManagerClientException Invalid notification template
+     */
+    public static void validateTemplateLocale(String locale) throws NotificationTemplateManagerClientException {
+
+        if (StringUtils.isBlank(locale)) {
+            String errorCode =
+                    I18nEmailUtil.prependOperationScenarioToErrorCode(
+                            I18nMgtConstants.ErrorMessages.ERROR_CODE_EMPTY_LOCALE.getCode(),
+                            I18nMgtConstants.ErrorScenarios.EMAIL_TEMPLATE_MANAGER);
+            throw new NotificationTemplateManagerClientException(errorCode,
+                    I18nMgtConstants.ErrorMessages.ERROR_CODE_EMPTY_LOCALE.getMessage());
+        }
+        // Regex check for registry invalid chars.
```

Comment:
When we move out of registry, these lines might not make sense. Though the allowed characters still makes sense, at some point we have to update the comments, variables and constants.

Reply:
Agreed.. Since the current implementation has a registry based impl, it will be better to keep this.. But this has to be removed when we totally remove the registry dependency. Will keep track this as a part of that task.